### PR TITLE
XP-479 Start training rounds from 0

### PR DESCRIPTION
**Pull request by finiteprods**
### References

[XP-404](https://xainag.atlassian.net/browse/XP-404) (specifically [XP-479](https://xainag.atlassian.net/browse/XP-479))

### Summary

This lays the groundwork for supporting weights initialisation in the Participant. Currently, the Coordinator is responsible for providing the initial weights to its participants for training, but it would be more desirable for this to be generated by the participants themselves.

For this, we adopt a new convention of enumerating rounds from **0** where the 0th round is usually intended to be for initialisation purposes but (for sake of simplicity and uniformity) is conceptually no different from any other round i.e. it still follows the same `StartTrainingRound`-`EndTrainingRound` communication:

- In `StartTrainingRound`, empty weights are distributed to participants.
- In `EndTrainingRound`, each participant uploads its share of initialised weights.

Thus the global initial weights for round 1 are the result of aggregating the shares from round 0.

Note that as a result of this change,

- `0 <= current_round < num_rounds`

### Are there any open tasks/blockers for the ticket?

- see also the related ticket [XP-381](https://xainag.atlassian.net/browse/XP-381)

---

### Reviewer checklist

*Reviewer agreement:*

* Reviewers assign themselves at the start of the review.
* Reviewers do **not** commit or merge the merge request.
* Reviewers have to check and mark items in the checklist.

**Merge request checklist**

- [x] Conforms to the merge request title naming `XP-XXX <a description in imperative form>`.
- [ ] Each commit conforms to the naming convention `XP-XXX <a description in imperative form>`.
- [x] Linked the ticket in the merge request title or the references section.
- [x] Added an informative merge request summary.

**Code checklist**

- [x] Conforms to the branch naming `XP-XXX-<a_small_stub>`.
- [x] Passed scope checks.
- [x] Added or updated tests if needed.
- [x] Added or updated code documentation if needed.
- [x] Conforms to Google docstring style.
- [x] Conforms to XAIN structlog style.

### CODE REVIEW

File: xain_fl/coordinator/coordinator.py [hunk 1]
```diff
@@ -367,7 +367,7 @@ def _handle_end_training_round(
             self.weights = self.aggregator.aggregate(self.round.get_weight_updates())
 
             # update the round or finish the training session
-            if self.current_round == self.num_rounds:
+            if self.current_round >= self.num_rounds - 1:
```
Comment (little-dude) [inline xain_fl/coordinator/coordinator.py]: Isn't this equivalent to
```python
if self.current_round > self.num_rounds:
```
?

Comment (finiteprods) [reply xain_fl/coordinator/coordinator.py]: no, but it is equivalent to
```python
if self.current_round > self.num_rounds - 2:
```

Comment (little-dude) [reply xain_fl/coordinator/coordinator.py]: oh :man_facepalming: I read this as `+ 1` instead of `- 1`, sorry.

Comment (janpetschexain) [reply xain_fl/coordinator/coordinator.py]: no, it is equivalent to `self.current_round > self.num_rounds - 2`.

**Pull request merged**